services/obs/obs-build-status.js

```diff
@@ -0,0 +1,69 @@
+import Joi from 'joi'
+
+// colors from OBS webinterface
+/* const statusmap = {
+	//Green
+	succeeded: "#008000",
+	// Blue
+	building: "#069",
+	// Cyan
+	scheduled: "#007a7c",
+	// Yellow
+	signing: "#f0ad4e",
+	finished: "#f0ad4e",
+	// Red
+	unresolvable: "#dc3545",
+	broken: "#dc3545",
+	failed: "#dc3545",
+	// Gray
+	disabled: "#6c757d",
+	blocked: "#6c757d",
+	// Orange
+	"scheduled-warning": "#a50",
+	unknown: "#a50"
+} */
+
+// shields colors
+const statusmap = {
+  // Green
+  succeeded: 'success',
+  // Blue
+  building: 'blue',
+  // Cyan
+  scheduled: '#007a7c',
+  // Yellow
+  signing: 'yellow',
+  finished: 'yellow',
+  // Red
+  unresolvable: 'critical',
+  broken: 'critical',
+  failed: 'critical',
+  // Gray
+  disabled: 'inactive',
+  blocked: 'inactive',
+  // Orange
+  'scheduled-warning': 'important',
+  unknown: 'lightgray',
+}
+
+const isBuildStatus = Joi.equal(...Object.keys(statusmap))
```

Comment:
I appreciate the desire to try to replicate the design of the native colors/presentation used upstream. However, our goal and design constraints are centered around consistency with _Shields_ and our badge specification, and it's pretty much a non-goal to be consistent with colors/presentation defined in an upstream service.

To that end of providing consistency across our badges, we have both schema helpers (`isBuildStatus`) and render helpers (`renderBuildStatusBadge`) both of which can be imported `build-status.js` and used. It looks like a third of the status values we may see in the OBS response are already accounted for (`succeeded`, `building`, `scheduled`, and `failed`).

`broken` isn't a status we track in the shared helpers yet, but it should definitely be added to our `redStatuses` list because it's an objectively clear/obvious failure indicator. I gather from context that `unresolvable` is a clear error condition as well for this service, but that's something we'll want to handle local to this service class vs. adding to the shared/global lists (it could feasibly mean something different for a different type of build service/platform)

For `disabled`, `blocked`, and `unknown`, we'd want to qualify those as our standard "other" statuses, which we display with the inactive color by default/no color specification needed. The other three (`scheduled-warning`, `signing` and `finished`) are probably fine to treat as orange statuses, though again local to this service.

For the schema validation, let's use the standard `isBuildStatus` validator but as part of an alternative set to include the other 7 (4 already covered in that validator plus `broken` being added as a known/shared red status)

e.g.
```js
Joi.alternatives()
    .try(isBuildStatus, Joi.equal('unknown'))
    .required()
```

Not that's just shared as a reference, the other 7 don't have to be enumerated individually with `Joi.equal`.

For rendering, we also want to rely on the standard/shared `renderBuildStatusBadge` helper as much as possible. There's really only 4 statuses we need to override the color for though (the 3 orange statuses mentioned above plus `unresolvable` as an error color), but for the other 8 we should let `renderBuildStatusBadge` handle everything (it'll take car of setting the color correctly for all the inactive statuses too).